src/state.py

```diff
@@ -16,46 +16,63 @@ def get_score(self):
                     for classes in set(list(i.buddies.keys())): 
                         if i.subjects_and_classes[classes] == self.students[numbers].subjects_and_classes[classes]: 
-                            score += 30 
+                            score += 30     #the person is at the same class of his friend
+                            alone = False   #Flag-- 
                         else: 
                             score -= 20
                         
-
+            #checking target class
             for position, j in enumerate(i.subjects_and_classes): 
                 if j in i.subject_targets.keys(): 
-                    # check if the student got the target class 
                     if i.subjects_and_classes[j] in i.subject_targets[j]:  
-                        score += 40
+                        score += 40         #if the person got the target class 
+                        no_target = False   #Flag--
                     else: 
                         score -= 30
 
+                #checking givin classes 
                 if j in i.subject_give_ins.keys():
-                    # check the givin classes but not so much
                     if i.subjects_and_classes[j] in i.subject_give_ins[j]: 
-                        score -= 3      #if a class was abdicated
+                        score -= 3          #if a class was abdicated
+                        no_givin = False    #Flag--
                     else: 
                         score += 5
                      
 
-               #checking schedule conflicts. If two classes are incompatible then score = -100000
+               #checking schedule conflicts. If two classes are incompatible then score = -inf 
                 for p in range(position+1, len(list(i.subjects_and_classes.keys()))):
                     key = list(i.subjects_and_classes.keys())[p] 
                     sched_1 = self.class_schedules[translate_subject_and_class(j,i.subjects_and_classes[j])]
                     sched_2 = self.class_schedules[translate_subject_and_class(key, i.subjects_and_classes[key])] 
                     if (sched_1.conflicts(sched_2)):
-                        score = -100000
+                        score = float('-inf')
                         # tests to print, please don't delete
                         # print("--Conflict between: {%s,%s}" %(name_1, name_2))
                         # print("----HOURS_1----")
                         # print("START: %i \n END_ %i" %(sched_1.start_hour.hours, sched_1.end_hour.hours))
                         # print("---HOURS_2----")
                         # print("START: %i \n END_ %i" %(sched_2.start_hour.hours, sched_2.end_hour.hours))
                         break 
-                               
+                
+            #case the person didn't get the target, didn't givin the class and is alone 
+            if alone and no_givin and no_target:    #once the score is -inf it will not change 
```

Comment:
This condition does not seem like something we need.
Let's say there's a student who does not get neither any of his targets nor gives anything in. 
Also, he did not chose any buddies.

This condition would make the state score -inf and this state may be acceptable.
At most, it's unacceptable because of some other student.

After changing the flags' scope to be an argument of each student you should enforce the following:
- Whenever some student changes to a class that is neither his target nor his give in nor a class that he has with a buddy, the score should be -infinity.
- When some student gives something in, if he does not get a at least one target, the score should be -infinity.

Reply:
So, basically this is the case in that the student didn't change his classes?